### PR TITLE
[Admin][improvement] Do not allow HTML tags for product attribute options when type is dropdown or multiple select

### LINKED ISSUE #29602: [Issue] [Admin][improvement] Do not allow HTML tags for product attribute options when type is dropdown or multiple select
This issue is automatically created based on existing pull request: magento/magento2#27417: [Admin][improvement] Do not allow HTML tags for product attribute options when type is dropdown or multiple select

---------
### Description (*)
This PR aims to add a new validation for the options of the product attribute type "Dropdown", "Multiple Select" that checks whenever the option value contains HTML tags.

HTML tags are not useful nor displayed on the storefront for the product attribute options, those the store manager should be aware of it.

### Related Pull Requests
https://github.com/magento/magento2/pull/27371

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)

1. Go to the Admin area
2. Go to Stores -> Attributes -> Product
3. Press the button "Add New Attribute"
4. Fill in the attribute default label
5. Select the "Catalog Input Type for Store Owner" to "Dropdown" or "Multiple Select"
6. In the section "Manage Options (Values of Your Attribute)" below add new options that contain HTML tags like `<span>`, `</strong>`, etc.
7. Press the "Save and Continue Edit" or "Save Attribute" button
8. See the error message
Error message that says the option contains HTML tags

### Questions or comments
Are the provided Unit and MFTF tests enough or there should be done more?

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by vasilii-b**
### Description (*)
This PR aims to add a new validation for the options of the product attribute type "Dropdown", "Multiple Select" that checks whenever the option value contains HTML tags.

HTML tags are not useful nor displayed on the storefront for the product attribute options, those the store manager should be aware of it.

### Related Pull Requests
https://github.com/magento/magento2/pull/27371

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)

1. Go to the Admin area
2. Go to Stores -> Attributes -> Product
3. Press the button "Add New Attribute"
4. Fill in the attribute default label
5. Select the "Catalog Input Type for Store Owner" to "Dropdown" or "Multiple Select"
6. In the section "Manage Options (Values of Your Attribute)" below add new options that contain HTML tags like `<span>`, `</strong>`, etc.
7. Press the "Save and Continue Edit" or "Save Attribute" button
8. See the error message
Error message that says the option contains HTML tags

### Questions or comments
Are the provided Unit and MFTF tests enough or there should be done more?

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29602: [Admin][improvement] Do not allow HTML tags for product attribute options when type is dropdown or multiple select

### CODE REVIEW

**Comment by vasilii-b** [comment]
@magento run all tests

**Comment by rogyar** [comment]
@magento run all tests

**Comment by rogyar** [comment]
Hi @vasilii-b. Considering the fact that the only static tests were executed we can assume that Magento cannot be installed after the change you've provided (theoretically). I would recommend doing the following for confirming this assumption. 

- Switch to the `2.4-develop` branch
- Pull the latest changes
- Merge your branch into the `2.4-develop` branch
- Run Magento installation with a clean database (via CLI)
- Make sure that Magento is installed successfully

**Comment by vasilii-b** [comment]
Hi @rogyar,
Thank you so much! I'll give it a try.

**Comment by sidolov** [comment]
@magento create issue

File: app/code/Magento/Eav/Model/Validator/Attribute/Options.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * Class Options
+ *
+ * Validation of EAV attribute options value
+ */
+class Options extends AbstractValidator
+{
+    /**
+     * Validation pattern for attribute options HTML tags
+     */
+    const VALIDATION_HTML_TAGS_RULE_PATTERN = '/<[^<]+>/';
```
Comment (sidolov) [inline app/code/Magento/Eav/Model/Validator/Attribute/Options.php]: Please, make const private

File: app/code/Magento/Eav/Model/Validator/Attribute/Options.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * Validation message for attribute options with HTML tags
+     */
+    const VALIDATION_HTML_TAGS_RULE_MESSAGE = 'HTML tags are not allowed for the attribute options. '.
```
Comment (sidolov) [inline app/code/Magento/Eav/Model/Validator/Attribute/Options.php]: Please, make const private

**Review by sidolov — APPROVED** [review]

**Review by sidolov — CHANGES_REQUESTED** [review]
Please, take a look at the review comments

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
### Before changes :heavy_check_mark:
I can create dropdown product attribute with:

- `Visible on Catalog Pages on Storefront` -> Yes
- `Allow HTML Tags on Storefront` -> Yes

And option value with Html tags which visible on PDP
image

### After changes :heavy_multiplication_x:
I try to create the same attribute and get the following error
image

@vasilii-b could you please take a look?
Thank you.

**Comment by engcom-Charlie** [comment]
Hi @vasilii-b, I'm closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for your collaboration.

**Pull request closed without merging**